AirCasting/SearchAndFollow/SearchMap/SearchMapViewModel.swift

```diff
@@ -113,26 +113,55 @@ class SearchMapViewModel: ObservableObject {
-                                        username: s.username,
-                                        uuid: s.uuid,
-                                        title: s.title,
                                         location: .init(latitude: s.latitude, longitude: s.longitude),
-                                        startTime: s.startTimeLocal,
-                                        endTime: s.endTimeLocal,
                                         markerImage: UIImage(systemName: "circle.circle.fill")!,
-                                        streamId: stream.id,
-                                        thresholdsValues: ThresholdsValue(veryLow: Int32(stream.thresholdVeryLow), low: Int32(stream.thresholdLow), medium: Int32(stream.thresholdMedium), high: Int32(stream.thresholdHigh), veryHigh: Int32(stream.thresholdVeryHigh)))
+                                        session: .init(id: s.id,
+                                                       uuid: s.uuid,
+                                                       provider: s.username,
+                                                       name: s.title,
+                                                       startTime: self.timeAsDate(s.startTimeLocal),
+                                                       endTime: self.timeAsDate(s.endTimeLocal),
+                                                       longitude: s.longitude,
+                                                       latitude: s.latitude,
+                                                       stream: s.streams.values.map { stream in
+                        .init(
+                            id: stream.id,
+                            unitName: stream.unitName,
+                            unitSymbol: stream.unitSymbol,
+                            measurementShortType: stream.measurementShortType,
+                            measurementType: stream.measurementType,
+                            sensorName: stream.sensorName,
+                            sensorPackageName: stream.sensorPackageName,
+                            thresholdsValues: .init(veryLow: Int32(stream.thresholdVeryLow),
+                                                    low: Int32(stream.thresholdLow),
+                                                    medium: Int32(stream.thresholdMedium),
+                                                    high: Int32(stream.thresholdHigh),
+                                                    veryHigh: Int32(stream.thresholdVeryHigh))
+                        )
+                })
+                )
             }
         }
     }
     
+    private func timeAsDate(_ time: String) -> Date {
+        let formatter = DateFormatters.SearchAndFollow.timeFormatter
+        let date = formatter.date(from: time)
+        guard let d = date else {
+            Log.error("Failed to convert start time received from API to date")
+            return DateBuilder.getFakeUTCDate()
+        }
+        return d
+    }
```

Comment:
This can go maybe to some extension?

Reply:
side note: it's a bit weird we have to use this anyway. Why don't we use a proper formatter for `JSONDecoder` on a service level and produce `Date` instead of `String`? Take a look at [docs](https://developer.apple.com/documentation/foundation/jsondecoder/2895216-datedecodingstrategy)